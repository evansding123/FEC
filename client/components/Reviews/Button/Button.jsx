import React from 'react';
import PropTypes from 'prop-types';

const Button = (props) => {
  const {
    moreReview,
    addReview,
    moreBTNshow,
  } = props;
  return (
    <div
      data-testid="addmoreBtn"
      className="add_moreBTNContainer"
    >
      {moreBTNshow === true
        ? (
          <button
            data-testid="moreBtn"
            className="add_moreBTN"
            type="button"
            onClick={moreReview}
          >
            MORE REVIEWS
          </button>
        ) : null }
      <button
        data-testid="AddBtn"
        className="add_moreBTN"
        type="button"
        onClick={addReview}
      >
        ADD A REVIEWS
      </button>
    </div>
  );
};

Button.propTypes = {
  moreBTNshow: PropTypes.bool,
  addReview: PropTypes.func,
  moreReview: PropTypes.func,
};

Button.defaultProps = {
  moreBTNshow: true,
  addReview: () => (1),
  moreReview: () => (1),
};

export default Button;
