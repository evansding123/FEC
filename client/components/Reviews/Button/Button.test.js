/**
* @jest-environment jsdom
*/
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  cleanup,
  fireEvent,
  screen,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import Button from './Button';

afterEach(cleanup);

describe('render correct', () => {
  it('when moreBTNshow be true addBTN should show', () => {
    const test1 = true;
    render(<Button moreBTNshow={test1} />);
    const test = screen.queryByText('MORE REVIEWS');
    expect(test).toBeInTheDocument();
  });

  it('when moreBTNshow be fale addBTN should not show', () => {
    const test2 = false;
    render(<Button moreBTNshow={test2} />);
    const test = screen.queryByText('MORE REVIEWS');
    expect(test).not.toBeInTheDocument();
  });
});

describe('click should trigger func', () => {
  const test1 = true;
  const test2 = false;
  it('when moreBTNshow be true, moreBTN should be able to trigger', () => {
    const testfunc = jest.fn();
    const { getByTestId } = render(
      <Button
        moreBTNshow={test1}
        moreReview={testfunc}
      />,
    );
    fireEvent.click(getByTestId('moreBtn'));
    expect(testfunc).toHaveBeenCalledTimes(1);
    fireEvent.click(getByTestId('moreBtn'));
    expect(testfunc).toHaveBeenCalledTimes(2);
  });

  it('when moreBTNshow be true, AddBTN should be able to trigger', () => {
    const testfunc = jest.fn();
    const { getByTestId } = render(
      <Button
        moreBTNshow={test1}
        addReview={testfunc}
      />,
    );
    fireEvent.click(getByTestId('AddBtn'));
    expect(testfunc).toHaveBeenCalledTimes(1);
    fireEvent.click(getByTestId('AddBtn'));
    expect(testfunc).toHaveBeenCalledTimes(2);
  });

  it('when moreBTNshow be false, AddBTN should be able to trigger', () => {
    const addReview = jest.fn();
    const { getByTestId } = render(
      <Button
        moreBTNshow={test2}
        addReview={addReview}
      />,
    );
    fireEvent.click(getByTestId('AddBtn'));
    expect(addReview).toHaveBeenCalledTimes(1);
    fireEvent.click(getByTestId('AddBtn'));
    expect(addReview).toHaveBeenCalledTimes(2);
  });
});

it('when moreBTNshow be true, AddBTN should be able to trigger', () => {
  const testfunc = jest.fn();
  const test = true;
  const { getByTestId } = render(
    <Button
      moreBTNshow={test}
      moreReview={testfunc}
    />,
  );
  fireEvent.click(getByTestId('moreBtn'));
  expect(testfunc).toHaveBeenCalledTimes(1);
});

test('should have default addReview and moreReview', () => {
  expect(Button.defaultProps.addReview()).toBeDefined();
  expect(Button.defaultProps.moreReview()).toBeDefined();
});

it('matches snapShot', () => {
  const test = true;
  const test1 = jest.fn();
  const test2 = jest.fn();
  const tree = renderer.create(
    <Button
      moreBTNshow={test}
      addReview={test1}
      moreReview={test2}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
