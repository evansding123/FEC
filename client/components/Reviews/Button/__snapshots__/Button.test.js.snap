// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="add_moreBTNContainer"
  data-testid="addmoreBtn"
>
  <button
    className="add_moreBTN"
    data-testid="moreBtn"
    onClick={[MockFunction]}
    type="button"
  >
    MORE REVIEWS
  </button>
  <button
    className="add_moreBTN"
    data-testid="AddBtn"
    onClick={[MockFunction]}
    type="button"
  >
    ADD A REVIEWS
  </button>
</div>
`;
